### DISCUSSION TITLE
chaining and  "filtering" datasets  within MODIS  h4 data

### DISCUSSION THREAD

**Discussion by lwasser** [corteva/rioxarray#267] [Q&A] [answered]
Hi again @snowman2 ,
The  clip  approaches that  we've discussed have worked  well  for  all data with the exception of  MODIS  h4 data. that  structure I believe has a main  dataset  and then  subdatasets. When I use  `rxr.open_raster("path-here.h4")` it returns  a  list  of two  xarray objects, a  qa  object  and  an object with the  spectral  "band"  data in it.

In this  instance where the hierarchical  data returns  two objects, is  there a  clever  way to  clip the  output data  that  takes  advantage  of  `from_disk`  (e.g.`rxr.open_raster("path-here.h4").clip(geometry,  from_disk=True)`  to  reduce memory and resource load? Right  now  we  have to  take each  object and clip individually as an error  is  returned when  I attempt  to  chain functions.  Many thanks!

@nkorinek just pinging  you here  too  so you  can  watch the thread too!

**Comment by snowman2** [comment, marked as answer]
You can use the variable/group kwargs to filter. See tests for examples:

https://github.com/corteva/rioxarray/blob/22251db2df18201f707ca5560bafb3def61d1212/test/integration/test_integration__io.py#L211

**Reply by lwasser** [reply]
that  did it!  thank you.  @snowman2 where did you  get the group  name  from (**MODIS_Grid_500m_2D**)? 

I also  noticed that i  can  just  grab the specific  bands that i  want using `variable=`  in  the  tests. That  also  works  great.
Thank you. that  will save a  lot of processing time.

**Reply by snowman2** [reply]
> where did you get the group name 

You can get it from inspecting the file (`gdalinfo` or by inspecting the `subdatasets` attribute on a rasterio dataset).
